### ISSUE TITLE
Performance data: r8169 support on kernel 4.13

### ISSUE THREAD

**Issue by harish-agr** [luigirizzo/netmap#502]
Hello,
I have found a issue on kernel version 4.13.0.19 with applying realtek r8169.c patch
./configure --kernel-sources=/home/dell/velocity/source/linux-4.13 --kernel-dir=/home/dell/velocity/source/linux-4.13/
generated output:
********************************** WARNING **********************************
*** 'r8169.c': no applicable patch found and no override provided.
*** Disabling driver.

Does it mean that we can not get any performance boost on r8169 + kernel 4.13 ???

**Comment by harish-agr** [comment]
Any body please comment on this....

**Comment by giuseppelettieri** [comment]
The generic driver should be sufficient to give you a performance boost for this card, measured by packet-rate. If you are looking for lower CPU utilisation, instead, you may have to actually try and see.

**Comment by harish-agr** [comment]
Thanks for reply. 

Do you mean that if I integrate netmap code with linux kernel 4.13.xx without using additional driver patch of r8169 then I can get performance boost.

**Comment by giuseppelettieri** [comment]
Yes, netmap will still work and you will get batching, zero-copy, etc. Only, the code path in the kernel will be a bit longer.

Il 11 Luglio 2018 11:09:33 CEST, harish-agr <notifications@github.com> ha scritto:
>Thanks for reply. 
>
>Do you mean that if I integrate netmap code with linux kernel 4.13.xx
>without using additional driver patch of r8169 then I can get
>performance boost. 
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/luigirizzo/netmap/issues/502#issuecomment-404099942

-- Inviato dal mio cellulare Android con K-9 Mail.

**Comment by RishiDarkDevil** [comment]
> Hello, I have found a issue on kernel version 4.13.0.19 with applying realtek r8169.c patch ./configure --kernel-sources=/home/dell/velocity/source/linux-4.13 --kernel-dir=/home/dell/velocity/source/linux-4.13/ generated output: ********************************** WARNING ********************************** *** 'r8169.c': no applicable patch found and no override provided. *** Disabling driver.
> 
> Does it mean that we can not get any performance boost on r8169 + kernel 4.13 ???

I tested the generic/emulated driver and native patched driver on Linux 6.11.13 for this driver on my machine (Ubuntu 24.10). I found the results exactly the same as what @giuseppelettieri mentioned
>The generic driver should be sufficient to give you a performance boost for this card, measured by packet-rate. If you are looking for lower CPU utilisation, instead, you may have to actually try and see.

I have added netmap native support for `r8169` by modifying the `configure` script, `if_re_netmap_linux.h` and adding the patch for the current version of the r8169 driver code provided in-tree with linux 6.11.13 and can be found [here](https://github.com/RishiDarkDevil/netmap/tree/master). 

Note: In my configure script the driver is referred to as `realtek` instead of `r8169`.
